src/lib/comms/sol-network-impl-contiki.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+
+#include <contiki-net.h>
+
+#include "sol-buffer.h"
+#include "sol-log.h"
+#include "sol-network.h"
+#include "sol-util.h"
+#include "sol-vector.h"
+
+static struct sol_vector links = SOL_VECTOR_INIT(struct sol_network_link);
+
+SOL_API const char *
+sol_network_link_addr_to_str(const struct sol_network_link_addr *addr,
+    struct sol_buffer *buf)
+{
+    char *p;
+    int i;
+    bool sep = false, skipping = false, treated_zeroes = false;
+
+    SOL_NULL_CHECK(addr, NULL);
+    SOL_NULL_CHECK(buf, NULL);
+
+    if (addr->family != SOL_NETWORK_FAMILY_INET6)
+        return NULL;
+
+    if (buf->capacity - buf->used < SOL_INET_ADDR_STRLEN)
+        return NULL;
+
+    p = sol_buffer_at_end(buf);
+
+    for (i = 0; i < 16; i += 2) {
+        uint16_t part;
+
+        part = ((uint16_t)addr->addr.in6[i] << 8) | addr->addr.in6[i + 1];
+        if (part && skipping) {
+            skipping = false;
+            treated_zeroes = true;
+            sep = true;
+        }
+        if (!part && !treated_zeroes) {
+            if (!skipping) {
+                skipping = true;
+                sep = true;
+            }
+        }
+        if (sep) {
+            sol_buffer_append_char(buf, ':');
+            if (skipping)
+                sep = false;
+        }
+        if (skipping)
+            continue;
+        sol_buffer_append_printf(buf, "%" PRIx16, part);
+        sep = true;
+    }
+    if (skipping)
+        sol_buffer_append_char(buf, ':');
+
+    return p;
```

Comment:
On Mon, 18 Apr 2016 11:44:48 -0700, Guilherme Íscaro wrote:
> > +            }
> > +        }
> > +        if (sep) {
> > +            sol_buffer_append_char(buf, ':');
> > +            if (skipping)
> > +                sep = false;
> > +        }
> > +        if (skipping)
> > +            continue;
> > +        sol_buffer_append_printf(buf, "%" PRIx16, part);
> > +        sep = true;
> > +    }
> > +    if (skipping)
> > +        sol_buffer_append_char(buf, ':');
> > +
> > +    return p;
> 
> Don't you need to check the return values of sol_buffer_append() functions?

I already checked at the beginning that there was enough space, what
other error there can be?